src/registry.cpp

```diff
@@ -149,13 +151,38 @@ uint32_t jitc_registry_id(const void *ptr) {
 
 uint32_t jitc_registry_id_bound(JitBackend backend, const char *domain) {
     Registry &r = registry;
+    if (!domain) {
+        uint32_t i = 0;
+        for (Domain &domain : r.domains) {
+            if (domain.backend == backend)
```

Comment:
I guess you could create a `DomainKey{ backend, domain_name }` to look up the domain ID into `r.domain_ids`, but I don't think it would be faster than your current implementation.

Reply:
Agreed. I guess it depends on how many domains there are with a different backend.

Reply:
I'll leave it like this for now.